### PR TITLE
inplace HardTanh, subclass ReLU6

### PR DESCRIPTION
**Pull request by szagoruyko**
as discussed in https://github.com/torch/nn/pull/849
* remove relu6 from THNN
* add inplace to HardTanh
* subclass ReLU6 from HardTanh

### CODE REVIEW

File: ReLU6.lua [hunk 1]
```diff
@@ -1,7 +1,9 @@
-local ReLU6, parent = torch.class('nn.ReLU6', 'nn.Module')
+local ReLU6, parent = torch.class('nn.ReLU6', 'nn.HardTanh')
```
Comment (soumith) [inline ReLU6.lua]: while this is the most effective approach (and i think so as well), after discussing with @jonathantompson , he was concerned that having a self.max_val in ReLU6 will make people believe that it's a tunable parameter.
Can you instead not subclass it but call the THNN.HardTanh.updateOutput functions with the value 6 directly?

Comment (szagoruyko) [inline ReLU6.lua]: @soumith done

File: HardTanh.lua [hunk 2]
```diff
@@ -14,7 +18,8 @@ function HardTanh:updateOutput(input)
       input:cdata(),
       self.output:cdata(),
       self.min_val,
-      self.max_val
+      self.max_val,
```
Comment (soumith) [inline HardTanh.lua]: backcompat? self.max_val or 1
self.inplace or false

Comment (szagoruyko) [inline HardTanh.lua]: added inplace, why self.max_val? it was there

**Comment by soumith** [comment]
Thanks Sergey!

**Pull request merged**